hardware_interface/src/resource_manager.cpp

```diff
@@ -514,9 +514,11 @@ class ResourceStorage
     {
       check_for_duplicates(hardware_info);
       load_hardware<Actuator, ActuatorInterface>(hardware_info, actuator_loader_, container);
-      initialize_hardware(hardware_info, container.back());
-      import_state_interfaces(container.back());
-      import_command_interfaces(container.back());
+      if (initialize_hardware(hardware_info, container.back()))
+      {
+        import_state_interfaces(container.back());
+        import_command_interfaces(container.back());
+      }
```

Comment:
```suggestion
      }
      else
      {
        RCUTILS_LOG_WARN_NAMED(
              "resource_manager", "hardware component named '%s' of type  '%s' failed to initialize.",
              hardware_info.name.c_str(), hardware_info.hardware_plugin_name.c_str());
      }
```

@destogl instead of tests, what do you think of this?

Reply:
This is anyway useful. And yes, it could be a solution anyway. Let me check.

Reply:
Yes/no?